business/dashboards.go

```diff
@@ -148,3 +169,30 @@ func (in *DashboardsService) GetIstioDashboard(params prometheus.IstioMetricsQue
 
 	return &dashboard, nil
 }
+
+func (in *DashboardsService) getDashboardTitle(namespace, template string) string {
+	key := dashboardKey(namespace, template)
+	if title, ok := dashboardTitles[key]; ok {
+		return title
+	}
+	dashboard, err := in.loadDashboardResource(namespace, template)
+	if err != nil {
+		log.Errorf("Cannot get dashboard %s in namespace %s", template, namespace)
+		return ""
+	}
+	return dashboard.Spec.Title
```

Comment:
Instead to return the dashboard.Spec.title, why not to fill the dashboardTitles and return always the cache ?

Reply:
`dashboardTitles` is filled anyway via `in.loadDashboardResource`, so it isn't really different from what you say. I can rename it to make it clearer it's updating the cache.